src/services/reviews.js

```diff
@@ -0,0 +1,362 @@
+						await kafkaCommunication.pushResourceToKafka(resource, resource.type)
+					} else {
+						//api need to implement
+					}
+				}
+
+				return responses.successResponse({
+					statusCode: httpStatusCode.ok,
+					message: 'RESOURCE_PUBLISHED',
+				})
+			}
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'REVIEW_UPDATED',
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async handleRejectionOrReport(
+		reviewId,
+		bodyData,
+		resourceId,
+		loggedInUserId,
+		comments = [],
+		resourceType,
+		resourceOrgId,
+		reviewOrgId
+	) {
+		try {
+			//program cannot reject or report
+			if (resourceType === common.RESOURCE_TYPE_PROGRAM) {
+				return responses.failureResponse({
+					message: 'PROGRAM_REJECTION_NOT_ALLOWED',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			// handle rejection or report of a review
+			let updateObj = { status: bodyData.status }
+			if (bodyData.notes) {
+				updateObj.notes = bodyData.notes
+			}
+
+			await resourceQueries.updateOne(
+				{ id: resourceId, organization_id: resourceOrgId },
+				_.omit(updateObj, ['notes'])
+			)
+
+			await reviewsQueries.update({ id: reviewId, organization_id: reviewOrgId }, updateObj)
+
+			// Add or update comments
+			if (comments) {
+				await handleComments(comments, resourceId, loggedInUserId)
+			}
+
+			const message =
+				bodyData.status === common.REVIEW_STATUS_REJECTED ? 'REVIEW_REJECTED' : 'REVIEW_REJECTED_AND_REPORTED'
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async handleChangesRequested(
+		reviewId,
+		bodyData,
+		resourceId,
+		loggedInUserId,
+		comments = [],
+		organization_id
+	) {
+		try {
+			//update reviews table
+			await reviewsQueries.update({ id: reviewId, organization_id: organization_id }, { status: bodyData.status })
+
+			// Add or update comments
+			if (comments) {
+				await handleComments(comments, resourceId, loggedInUserId)
+			}
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'REVIEW_CHANGES_REQUESTED',
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async handleApproval(
+		reviewId,
+		resourceId,
+		loggedInUserId,
+		comments = [],
+		minApproval,
+		resourceOrgId,
+		reviewOrgId
+	) {
+		try {
+			// handle approval of a review
+			let publishResource = false
+			await reviewsQueries.update(
+				{ id: reviewId, organization_id: reviewOrgId },
+				{ status: common.RESOURCE_STATUS_APPROVED }
+			)
+
+			// Add or update comments
+			if (comments) {
+				await handleComments(comments, resourceId, loggedInUserId)
+			}
+
+			//check the no of approvals meets
+			const reviewsApproved = await reviewsQueries.countDistinct({
+				resource_id: resourceId,
+				status: common.REVIEW_STATUS_APPROVED,
+			})
+
+			if (minApproval <= reviewsApproved + 1) {
```

Comment:
if minApproval of a resource is 2 and a reviewer approves this resource. then
reviewsApproved = 1
reviewsApproved + 1 = 2

then minApproval <= reviewsApproved + 1 will be true and it will be published right? 
@priyanka-TL